src/partition.rs

```diff
@@ -857,24 +862,23 @@ pub fn get_intra_edges<T: Pixel>(
       let dc_or_cfl =
         mode == PredictionMode::DC_PRED || mode == PredictionMode::UV_CFL_PRED;
 
-      needs_left = mode != PredictionMode::V_PRED && (!dc_or_cfl || x != 0)
+      needs_left = mode != PredictionMode::V_PRED && (!dc_or_cfl || left_edge != 0)
         && !(mode == PredictionMode::D45_PRED || mode == PredictionMode::D63_PRED);
       needs_topleft = mode == PredictionMode::PAETH_PRED || mode == PredictionMode::D117_PRED
       || mode == PredictionMode::D135_PRED || mode == PredictionMode::D153_PRED;
-      needs_top = mode != PredictionMode::H_PRED && (!dc_or_cfl || y != 0);
+      needs_top = mode != PredictionMode::H_PRED && (!dc_or_cfl || top_edge != 0);
       needs_topright = mode == PredictionMode::D45_PRED || mode == PredictionMode::D63_PRED;
       needs_bottomleft = mode == PredictionMode::D207_PRED;
     }
 
     // Needs left
     if needs_left {
-      if x != 0 {
-        let left_slice = dst.go_left(1);
+      if left_edge != 0 {
```

Comment:
Shouldn't this be "x != left_edge" ? I think current code will only work for 1st tile.
If we start to use multiple tiles and x and y come from "dst_content = dst.without_edges();",
then I think the border condition check here and all below are wrong.
We also needs asserts to check x >= left_edge and y >= top_edge?

Can you explain why not using 
x = dst.x; and  y = dst.y; 
given that, "dst: &EdgedPlaneSlice"?

Reply:
> Shouldn't this be "x != left_edge" ?

`left_edge` represents the number of pixels in the left edge.